frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/PushAction.java

```diff
@@ -0,0 +1,49 @@
+package com.virtuslab.gitmachete.frontend.actions;
+
+import com.intellij.dvcs.push.ui.VcsPushDialog;
+import com.intellij.dvcs.repo.VcsRepositoryManager;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ * </ul>
+ */
+public class PushAction extends AnAction implements DumbAware {
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent e) {
+    super.update(e);
+
+    Project project = e.getProject();
+    Option<GitRepository> selectedVcsRepository = ActionUtils.getSelectedVcsRepository(e);
+
+    boolean isSomeRepositoryPresent = project != null && !VcsRepositoryManager.getInstance(project).getRepositories().isEmpty();
```

Comment:
Hmm why do we need this check? Do we need this in any other action as well?

Reply:
good point, it is redundant